### PR TITLE
avocado.core.xunit: Escape non-printable characters

### PR DESCRIPTION
**Pull request by ldoktor**
Xunit/XML CDATA/attributes must not contain non-printable characters.
This patch escapes such characters using hex value.

Signed-off-by: Lukáš Doktor <ldoktor@redhat.com>

### CODE REVIEW

**Comment by ldoktor** [comment]
This patch fixes an issue with tests producing colored output (eg. kvm-unit-test)

```
This page contains the following errors:

error on line 9 at column 24: Input is not proper UTF-8, indicate encoding !
Bytes: 0x1B 0x5B 0x33 0x31
Below is a rendering of the page up to the first error.

11:01:35 DEBUG| [stdout] [31mFAIL[0m emulator
-------------------------^
```

File: avocado/core/xunit.py [hunk 1]
```diff
@@ -24,6 +25,9 @@
 # We use a subset of the XML format defined in this URL:
 # https://svn.jenkins-ci.org/trunk/hudson/dtkit/dtkit-format/dtkit-junit-model/src/main/resources/com/thalesgroup/dtkit/junit/model/xsd/junit-4.xsd
 
+PRINTABLE = string.ascii_letters + string.digits + string.punctuation + '\n\r '
```
Comment (apahim) [inline avocado/core/xunit.py]: Why not to use only `strings.printable`? It's almost the same of all above together, but it also includes all characters that are considered whitespace:

    >> PRINTABLE
    'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789!"#$%&\'()*+,-./:;<=>?@[\\]^_`{|}~\n\r '
    >> strings.printable
    '0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ!"#$%&\'()*+,-./:;<=>?@[\\]^_`{|}~ \t\n\r\x0b\x0c'

File: avocado/core/xunit.py [hunk 2]
```diff
@@ -34,16 +38,14 @@ def __init__(self):
         self.xml = ['<?xml version="1.0" encoding="UTF-8"?>']
 
     def _escape_attr(self, attrib):
+        attrib = ''.join(_ if _ in PRINTABLE else "\\x%02x" % ord(_)
+                         for _ in str(attrib))
```
Comment (apahim) [inline avocado/core/xunit.py]: cool.

File: avocado/core/xunit.py (hunk 1 above)
Comment (ldoktor) [inline avocado/core/xunit.py]: The `\x0b` and `\x0c' are not supported in xunit

Comment (apahim) [inline avocado/core/xunit.py]: Ok then.

**Comment by apahim** [comment]
Looks good.

**Pull request merged**